### PR TITLE
HackStudio: Add "Go Back" and "Go Forward" navigation actions 

### PR DESCRIPTION
**Pull request by itamar8910**
These actions allow the user to move backwards & forwards between previous project locations they were at.
It's something I find myself using quite a lot in CLion & was missing in HackStudio.

### CODE REVIEW

**Comment by alimpfard** [comment]
Nice!

Though it looks like the shortcuts (alt+left/alt+right) don't work, it just moves the cursor for me here.

**Comment by itamar8910** [comment]
Weird, the `alt` shortcuts work for me here. Maybe it's a qemu thing?
Anyways, does alt+shift+left (`{ Mod_Alt | Mod_Shift, Key_Left }`) work for you? It's what CLion has by default.

**Comment by alimpfard** [comment]
:thinking: it might be qemu or my window manager, I've never had alt+dot work in the terminal either...
I'll try that when I have access to my computer.

File: Userland/DevTools/HackStudio/HackStudioWidget.h [hunk 1]
```diff
@@ -138,6 +151,11 @@ class HackStudioWidget : public GUI::Widget {
 
     OwnPtr<Project> m_project;
 
+    Vector<ProjectLocation> m_locations_history;
+    Vector<ProjectLocation> m_locations_history_forward_list;
+    Optional<ProjectLocation> m_last_location;
+    bool m_locations_history_disabled { false };
```
Comment (awesomekling) [inline Userland/DevTools/HackStudio/HackStudioWidget.h:157]: This feels like an unnecessarily difficult way to store the history.
Can we store it in a single vector and have a cursor index pointing at the current item?

Comment (itamar8910) [reply Userland/DevTools/HackStudio/HackStudioWidget.h]: Yeah, this does simplify things. Done :+1:

**Pull request merged**